lib/blocks.php

```diff
@@ -296,3 +296,219 @@ function gutenberg_enqueue_editor_block_styles_assets() {
 	}
 	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_editor_block_styles_assets' );
 }
+
+/**
+ * Renders the classNames and styles for blocks
+ *
+ * @param string $block_content Output of the current block.
+ * @param array  $block Block Object.
+ * @return string New block output.
+ */
+function gutenberg_experimental_apply_classnames_and_styles( $block_content, $block ) {
+	// Don't filter template part blocks since we filter the blocks in each template
+	// part individually.
+	if ( 'core/template-part' === $block['blockName'] ) {
```

Comment:
Are there other blocks which work like this? Is this true for any block with controlled inner blocks? why is it an exception here but not for e.g. post content?

Reply:
Actually, I think this early return may be redundant.  That is, none of these blocks we are concerned with have (or are meant to ever have) the supports flags in question.  In which case they will all return early given the empty supports array.

I think @apeatling might have added this part?  Are there any reasons to keep this blacklist and include the other blocks Noah mentions?

Reply:
Let me retest this.